contracts/pool/src/tests/liquidity.rs

```diff
@@ -698,3 +700,281 @@ fn withdraw_liqudity_below_min() {
     pool.withdraw_liquidity(&user1, &share_amount, &3000, &3000);
 }
+
+#[test]
+fn query_share_valid_liquidity() {
+    let env = Env::default();
+    env.mock_all_auths();
+    env.budget().reset_unlimited();
+
+    let mut admin1 = Address::generate(&env);
+    let mut admin2 = Address::generate(&env);
+
+    let mut token1 = deploy_token_contract(&env, &admin1);
+    let mut token2 = deploy_token_contract(&env, &admin2);
+    if token2.address < token1.address {
+        std::mem::swap(&mut token1, &mut token2);
+        std::mem::swap(&mut admin1, &mut admin2);
+    }
+    let user1 = Address::generate(&env);
+    let user2 = Address::generate(&env);
+    let user3 = Address::generate(&env);
+    let stake_manager = Address::generate(&env);
+    let stake_owner = Address::generate(&env);
+
+    let pool = deploy_liquidity_pool_contract(
+        &env,
+        None,
+        (&token1.address, &token2.address),
+        0i64,
+        None,
+        None,
+        None,
+        stake_manager,
+        stake_owner,
+    );
+
+    let share_token_address = pool.query_share_token_address();
+    let token_share = token_contract::Client::new(&env, &share_token_address);
+
+    token1.mint(&user1, &150);
+    token1.mint(&user2, &250);
+    token1.mint(&user3, &350);
+
+    token2.mint(&user1, &200);
+    token2.mint(&user2, &300);
+    token2.mint(&user3, &400);
+
+    pool.provide_liquidity(&user1, &Some(150), &Some(10), &Some(200), &Some(10), &None);
+    pool.provide_liquidity(&user2, &Some(150), &Some(50), &Some(200), &Some(50), &None);
+    pool.provide_liquidity(
+        &user3,
+        &Some(150),
+        &Some(100),
+        &Some(200),
+        &Some(100),
+        &None,
+    );
+
+    // user1 assertions
+    let lp_share_balance_user1 = token_share.balance(&user1);
+    let query_share_result_user1 = pool.query_share(&lp_share_balance_user1);
+    assert_eq!(
+        query_share_result_user1,
+        (
+            Asset {
+                address: token1.address.clone(),
+                amount: 149
+            },
+            Asset {
+                address: token2.address.clone(),
+                amount: 199
+            }
+        )
+    );
+
+    pool.withdraw_liquidity(&user1, &lp_share_balance_user1, &100i128, &100i128);
+
+    let lp_share_balance_after_withdraw_user1: i128 = token_share.balance(&user1);
+    assert_eq!(lp_share_balance_after_withdraw_user1, 0);
+
+    let pool_info_result = pool.query_pool_info();
+    assert_eq!(
+        pool_info_result,
+        PoolResponse {
+            asset_a: Asset {
+                address: token1.address.clone(),
+                amount: 301
+            },
+            asset_b: Asset {
+                address: token2.address.clone(),
+                amount: 401
+            },
+            asset_lp_share: Asset {
+                address: share_token_address.clone(),
+                amount: 346i128
+            },
```

Comment:
That assertion makes sense if you compare it with the balance before withdrawing.

Reply:
alright, made a query for the pool info before and after the withdrawal and now I'm comparing that. After the final withdrawal we assert that evyrhing in pool_info is at 0